src/main/java/data/Camera.java

```diff
@@ -163,4 +163,17 @@ public boolean equals(Object o) {
             return camera.ip == null;
         }
     }
+    
+    /**
+     * Construct string to display this camera in a modal.
+     * @return the correct string
+     */
+    public String constructModalString() {
```

Comment:
I see this method 3 times exactly the same, could you maybe extract it and make it a static method or something?